features/ui/button/button.tsx

```diff
@@ -1,6 +1,45 @@
-import styled from "styled-components";
+import { color, space, textFont } from "@styles/theme";
+import styled, { css } from "styled-components";
 
-export const Button = styled.button`
+export enum ButtonSize {
+  sm = "sm",
+  md = "md",
+  lg = "lg",
+  xl = "xl",
+}
+
+export enum ButtonColor {
+  primary = "primary",
+  secondary = "secondary",
+  empty = "empty",
+  emptyGray = "emptyGray",
+  gray = "gray",
+  error = "error",
+}
+
+export enum IconDisplay {
+  none = "none",
+  only = "only",
+  leading = "leading",
+  trailing = "trailing",
+}
+
+// These props have been grouped into a single object because it would'nt make sense to ever specify only one of them
+export interface IconOptions {
+  src: string;
+  display: IconDisplay;
+}
+
+interface ButtonProps {
```

Comment:
For components in UI libraries it's often best to use unions of custom prop types with existing HTML element types. Here for example you can write

export type ButtonContainerProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  icon?: IconOptions;
  size: ButtonSize;
  color: ButtonColor;
};

I'd probably replace the `text` prop with `children`. The `children` and `disabled` props are already defined on the `ButtonHTMLAttributes<HTMLButtonElement>` type. 

This way you decrease the risk that you forget an important prop that another developer has to add later on. Like here for example you missed the `onClick` prop 😄 

When you define the prop type like this you can define the component like this:

```
export function Button({
  icon,
  size = ButtonSize.md,
  color = ButtonColor.primary,
  ...buttonProps
}: ButtonProps) {
  return (
    <Container
      size={size}
      color={color}
      iconDisplay={icon ? icon.display : IconDisplay.none}
     {...buttonProps}
    >
```

Reply:
Thank you so much for this tip, this is *exactly* what I was trying to achieve originally but could not figure it out. Cleans up the props nicely and covers all the usual attributes you would want. Really this is a godsend.